lib/guardian/post_guardian.rb

```diff
@@ -239,10 +239,17 @@ def can_see_post?(post)
     return false unless can_see_post_topic?(post)
     return false unless post.user == @user || Topic.visible_post_types(@user).include?(post.post_type)
     return true if is_moderator? || is_category_group_moderator?(post.topic.category)
-    return true if post.deleted_at.blank? || (post.deleted_by_id == @user.id && @user.has_trust_level?(TrustLevel[4]))
+    return true if !post.trashed? || can_see_deleted_post?(post)
     false
   end
 
+  def can_see_deleted_post?(post)
+    return false if !post.trashed?
```

Comment:
I got a little bit weirded out that this method also does a soft check that the post is actually deleted.

Reply:
I mainly put that in there to stop from security bugs coming out of people using this method on a post that is not deleted, since its more lax than `can_see_post?`, I do agree its weird tho

Reply:
Yeah. It's becomes a bit of a question of how defensive to be. I tend to make the assumption that callers are using the correct arguments (unless I'm writing library code), but that doesn't mean that's universally good. Security-related code might be a good exception. 🙂